website/addons/github/static/githubFangornConfig.js

```diff
@@ -187,7 +154,46 @@ var _githubItemButtons = {
                     }, branchArray)
                 );
             }
-        } else if (item.kind === 'file') {
+            if (tb.options.placement !== 'fileview') {
+                // If File and FileRead are not defined dropzone is not supported and neither is uploads
+                if (window.File && window.FileReader && item.data.permissions && item.data.permissions.edit) {
+                    buttons.push(
+                        m.component(Fangorn.Components.button, {
+                            onclick: function (event) {
+                                Fangorn.ButtonEvents._uploadEvent.call(tb, event, item);
+                            },
+                            icon: 'fa fa-upload',
+                            className: 'text-success'
+                        }, 'Upload'),
+                        m.component(Fangorn.Components.button, {
+                            onclick: function (event) {
+                                tb.toolbarMode(Fangorn.Components.toolbarModes.ADDFOLDER);
+                            },
+                            icon: 'fa fa-plus',
+                            className: 'text-success'
+                        }, 'Create Folder')
+                    );
+                }
+                if (item.data.addonFullname && tb.options.placement !== 'fileview') {
```

Comment:
This check for `tb.options.placement !== 'fileview'` is redundant, since you're checking it in an outer `if`.